src/backend/dx11/src/internal.rs

```diff
@@ -708,7 +708,33 @@ impl Internal {
         } else {
             // Default copy path
             for region in regions.into_iter() {
-                let info = region.borrow();
+                let info: &command::ImageCopy = region.borrow();
+
+                assert_eq!(
+                    src.decomposed_format.typeless,
+                    dst.decomposed_format.typeless,
+                    "DX11 backend cannot copy between underlying image formats: {} to {}.",
+                    src.decomposed_format.typeless,
+                    dst.decomposed_format.typeless,
+                );
+
+                let src_depth_stencil = src.format.is_depth() || src.format.is_stencil();
+                let dst_depth_stencil = dst.format.is_depth() || dst.format.is_stencil();
+
+                assert_eq!(src_depth_stencil, dst_depth_stencil, "Both images in a copy must be depth stencil textures");
+
+                let copy_box = if src_depth_stencil || dst_depth_stencil {
```

Comment:
It would be great to issue a warning here if the given `src_offset and `extent` don't cover the whole image. Otherwise, somebody in the future may waste time trying to figure out why it's not respected for the copy.